infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/CorfuLogReplicationRuntime.java

```diff
@@ -295,6 +296,11 @@ public synchronized void resetRemoteLeaderNodeId() {
         return connectedNodes;
     }
 
+    public synchronized void updateRouterClusterDescriptor(ClusterDescriptor clusterDescriptor) {
```

Comment:
the method is under 'synchronized' qualifier however, I see the calls from a for loop above. Is there a rpc-sort of communication involved to invoke this method?